pkg/api/metadata.go

```diff
@@ -134,3 +134,28 @@ func BuildCacheFor(metadata Metadata) ImageStreamTagReference {
 func ImageVersionLabel(fromTag PipelineImageStreamTagReference) string {
 	return fmt.Sprintf("io.openshift.ci.from.%s", fromTag)
 }
+
+var testPathRegex = regexp.MustCompile(`(?P<org>[^/]+)/(?P<repo>[^@]+)@(?P<branch>[^:]+):(?P<test>.+)`)
+
+func MetadataTestFromString(input string) (*MetadataWithTest, error) {
+	var ret MetadataWithTest
+	match := testPathRegex.FindStringSubmatch(input)
+	if match == nil || len(match) != 5 {
+		return &ret, fmt.Errorf("test path not in org/repo@branch:test or org/repo@branch__variant:test format: %s", input)
+	}
+	ret.Org = match[1]
+	ret.Repo = match[2]
+	ret.Test = match[4]
+
+	branchAndVariant := strings.Split(match[3], "__")
```

Comment:
Branches can contain `__` - not sure it matters to handle it, but also why IMO if you're going to stringify some data it's better to do it to a spec.

Reply:
It's the same specification we use in ci-op naming convention though, so afaik we'll break there already for branches that contain `__`.